{{cookiecutter.app_name}}/Makefile

```diff
@@ -1,19 +1,38 @@
-.PHONY: env-setup env-teardown dev install-dependencies test
+.PHONY: env-setup env-teardown db/migrate db/rollback migratoin/create migration/status dev install-dependencies test
 
 env-setup:
 	docker-compose -f docker-compose.dev.yml up -d
 
 env-teardown:
 	docker-compose -f docker-compose.dev.yml down
 
-dev: env-setup
+db/migrate:
+	goose -dir database/migrations -table "migration_versions" postgres "postgres://postgres:postgres@0.0.0.0:${DB_PORT}/test-gin-template_${ENV}?sslmode=disable" up
+
+db/rollback:
+	goose -dir database/migrations -table "migration_versions" postgres "postgres://postgres:postgres@0.0.0.0:$(DB_PORT)/test-gin-template_${ENV}?sslmode=disable" down
```

Comment:
@olivierobert According to slack conversation. I decided to add make command to wrap goose command. Goose migration command needs a database URL in order to connect to the database and migrate it. In this case I need ENV which I can pass from make test and make dev command but I think DB PORT should come from config.

I'm still struggling with how to implement this properly.

Reply:
> Another way I can think of is to create function in Go and execute goose command in bootstrap and use this in the setup step for both development and test. This way we can use config from .toml file using Viper.

It's possible and it could help to remove some of the complexity in the Makefile. However, I also think the current implementation would be okay.

Your blocker is that you feel an issue with the duplication of configuration variables in the `toml` and OS ENV. As I explained, it's not an issue. All applications use a mix of environment variables and static application configuration. APP_EMV, DB_HOST, DB_USER, DB_PASSWORD should be defined as env variables in production to make it easy for configuring servers but for dev/test environments having them as static configurations is better and easier.